outputs.tf

```diff
@@ -51,10 +55,15 @@ output "codebuild_badge_url" {
 
 output "codepipeline_id" {
   description = "CodePipeline ID"
-  value       = join("", aws_codepipeline.default.*.id)
+  value       = lookup(local.codepipeline_resource, "id", "")
```

Comment:
Are you sure this is equivalent? What happens with `lookup` is called on `null`?

Reply:
you're right - null is an invalid input to lookup(). Using a fallback value of `{}` in the try at the top (instead of null) seems to work OK.